pytensor/tensor/basic.py

```diff
@@ -3831,6 +3832,23 @@ def __setstate__(self, state):
             self.axis2 = 1
 
 
+class AllocDiag2(OpFromGraph):
```

Comment:
Why 2?

Reply:
Because there's `AllocDiag` legacy code that isn't used anywhere in the code base that warns you to use `alloc_diag` instead. My preference would be to delete that, but I was being cautious.

Reply:
Seems like a good time to delete it